__tests__/hathorwallet.test.js

```diff
@@ -32,5 +43,133 @@ test('Protected xpub wallet methods', async () => {
   await expect(hWallet.prepareMeltTokensData()).rejects.toThrow(WalletFromXPubGuard);
   await expect(hWallet.prepareDelegateAuthorityData()).rejects.toThrow(WalletFromXPubGuard);
   await expect(hWallet.prepareDestroyAuthorityData()).rejects.toThrow(WalletFromXPubGuard);
-  await expect(hWallet.getAllSignatures).toThrow(WalletFromXPubGuard);
+  expect(hWallet.getAllSignatures).toThrow(WalletFromXPubGuard);
+  expect(hWallet.getSignatures).toThrow(WalletFromXPubGuard);
+});
+
+test('getSignatures', () => {
+  // Spy on lib methods used
+  const mockStorageGet = jest.spyOn(Storage.prototype, 'getItem').mockReturnValue({
+    mainKey: 'mocked-encrypted-privkey',
+  });
+  const mockPrivkey = jest.spyOn(wallet, 'decryptData').mockReturnValue((new HDPrivateKey()).toString());
+  const mockInputData = jest.spyOn(transaction, 'createInputData');
+  const mockSetData = jest.spyOn(Input.prototype, 'setData');
+
+  // Setup transaction to test signature method
+  const tokenUid = '00034a15973117852c45520af9e4296c68adb9d39dc99a0342e23cd6686b295e';
+  const inputTx = '000164e1e7ec7700a18750f9f50a1a9b63f6c7268637c072ae9ee181e58eb01b';
+  const mockAddresses = ['WYBwT3xLpDnHNtYZiU52oanupVeDKhAvNp', 'WYiD1E8n5oB9weZ8NMyM3KoCjKf1KCjWAZ'];
+  const mockAddr0 = new Address(mockAddresses[0]);
+  const script0 = new P2PKH(mockAddr0);
+  const mockAddr1 = new Address(mockAddresses[1]);
+  const script1 = new P2PKH(mockAddr1);
+  const outputs = [
+    new Output(10, script0.createScript(), { tokenData: 1 }), // token funds
+    new Output(1, script0.createScript(), { tokenData: 129 }), // Mint authority
+    new Output(20, script1.createScript()), // HTR
+  ];
+  const tx0 = new Transaction(
```

Comment:
I think it would be helpful to have a comment explaining what are the objectives that the test on this tx will have.
In this case, would it be to identify that only the correct inputs are signed?

Reply:
Added the explanations